File: LeetCode/Easy/0557-reverse-words-in-a-string-iii/0557-reverse-words-in-a-string-iii.java
class Solution {
    public String reverseWords(String s) {

        StringBuilder sb = new StringBuilder();

        String[] sAry = s.split(" ");
        
        for(int j = 0; j < sAry.length; ++j) {
            char[] cAry = sAry[j].toCharArray();

            for(int i = cAry.length - 1; i >= 0; --i) {
                sb.append(cAry[i]);
            }
            if(j != sAry.length - 1) sb.append(" ");
        }

        return sb.toString();
    }
}

File: LeetCode/Easy/0557-reverse-words-in-a-string-iii/0557-reverse-words-in-a-string-iii.kt
class Solution {
    fun reverseWords(s: String): String {
        val sb: StringBuilder = StringBuilder()

        val sAry = s.split(" ")

        for( (i, str) in sAry.withIndex()) {
            val cAry = str.toCharArray()

            for ( j in cAry.lastIndex downTo 0) {
                sb.append(cAry[j])
            }

            if (i != sAry.lastIndex) sb.append(" ");
        }

        return sb.toString()
    }
}